geode-core/src/main/java/org/apache/geode/internal/net/NioSslEngine.java

```diff
@@ -402,6 +436,10 @@ public synchronized void close(SocketChannel socketChannel) {
           socketChannel.write(myNetData);
         }
       }
+      if (result != null && result.getStatus() != CLOSED) {
+        throw new SSLHandshakeException(
+            "Error closing SSL session.  Status=" + result.getStatus());
+      }
```

Comment:
Not thrilled with having to null-check here. See the comment in the loop body around line 424 for an explanation of why this rearrangement was necessary.

Reply:
I don't see why the rearrangement is necessary from the comment. I think you need to clarify. Is it because the write additional net data after close is required but if we exit the loop not in a closed state we are borked?